bitcoin/src/consensus/encode.rs

```diff
@@ -136,6 +136,21 @@ pub fn deserialize_partial<T: Decodable>(data: &[u8]) -> Result<(T, usize), Erro
     Ok((rv, consumed))
 }
 
+/// A trait for integer types smaller than or equal to 64-bits.
+///
+/// We have this trait here because [usize] doesn't implement `Into<u64>`.
+pub trait IntoU64 {
+    /// Convert into a [u64].
+    fn into_u64(self) -> u64;
+}
+
+impl IntoU64 for u8 { fn into_u64(self) -> u64 { self as u64 } }
+impl IntoU64 for u16 { fn into_u64(self) -> u64 { self as u64 } }
+impl IntoU64 for u32 { fn into_u64(self) -> u64 { self as u64 } }
+impl IntoU64 for u64 { fn into_u64(self) -> u64 { self } }
```

Comment:
In e9123d1d25bce9be4d1702ae9da6b99bd2e9674e:

It doesn't really matter but I kinda prefer using `.into` rather than casting for these cases, since it's possible.